### PR TITLE
Joram's Pull Request

### PR DESCRIPTION
**Pull request by swoleengineer**
Added route to delete all closed PRs

### CODE REVIEW

**Review by phytertek — APPROVED** [review]
I removed the merge conflicts in app.js for the db uri and in package.json for the start script, everything else looks good to go :)

**Comment by swoleengineer** [comment]
Did some refactoring in this branch, explained in docs/PROJECT_STRUCTURE...
Also added a function in the controller to handle when comments are made on diffs in pull requests.

It's a good idea to create multiple webhooks in the repositories, and just assign a specific endpoint for each webhook. Have that webhook be responsible for just one of the many options. The 'diff comments' alone returns a post with at least 3 options ('create', 'edit', 'delete)...

So I added a function that takes the post, sees what the action was and handles the comment from that point by either pushing it into the pull request, removing it, or editing it.

Also refactored the model for the pull requests to be able to accept these file comments.

File: api/home/home.controller.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const util = require('../utils/utilities');
+
+module.exports = () => {
+  return {
+		home: (req, res) => {
+			console.log('Found it!');
```
Comment (ryan-hamblin) [inline api/home/home.controller.js]: Indentation in this file is a bit too much for my tastes. Can we set a standard of 2 spaces per nested line?

**Review by ryan-hamblin — COMMENTED** [review]
I'm not sure I'm a big fan of naming files with a `.`. This is of course stylistic in preference, but I propose we make a standard and stick to it. I would prefer we do with camelCase or kebab-case for file nameing. So `pr.controller.js` would in turn become `prController.js` or `pr-controller.js`

File: api/pullrequests/pr.controller.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            // if PR exists we're going to just update it.
+            const id = objectFound[0]._id;
+            handleUpdatePullRequest(id, newPr, res, done);
+          }
+        });
+      } else {
+        res.status(200);
+        res.json({ 'Message: ': 'must be a test' });
+      }
+		},
```
Comment (ryan-hamblin) [inline api/pullrequests/pr.controller.js]: Indentation is off here.

File: api/pullrequests/pr.controller.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+		
+    getAllPrs: (req, res, done) => {
+			// console.log('GET ALL PRS');
+			PullRequest.find({}, (err, pullRequests) => {
```
Comment (ryan-hamblin) [inline api/pullrequests/pr.controller.js]: Lets pull these lines back in.

File: api/pullrequests/pr.controller.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+          avatar_url: req.body.comment.user.avatar_url
+        }
+      }
+
+      const deleteDiffComment = () => {
+        PullRequest.update({id: prId}, 
+          { $pull: { diffComments: { ghId }}}, 
+          { safe: true, multi: true }, 
+          (err, response) => {
+            return err ? util.errorHandler(500, 'Server error deleting this comment from the pull request.', res, done) : res.json(response);
```
Comment (ryan-hamblin) [inline api/pullrequests/pr.controller.js]: Love this style of err handling.

File: api/pullrequests/pr.controller.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+      const editDiffComment = () => {
+        PullRequest.update({id: prId, 'diffComments': {'$elemMatch': {ghId } }}, 
+          { $set : postComment }, 
+          { new:true, upsert:true, safe:true }, 
+          (err, response) => {
+            return err ? util.errorHandler(500, 'Error updating a diff comment for the pull request', res, done) : res.json(response);
+          })
+      }
+
+      switch(action) {
```
Comment (ryan-hamblin) [inline api/pullrequests/pr.controller.js]: Do these actions come from the Hook event submitted from Github?!?! How have I not seen this!?

File: api/pullrequests/pr.model.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+const PrSchema = new Schema({
+    url: String,
+    state: String,
+    name: String,
+    createdAt: Date,
+    id: Number,
+    diffComments: [diffCommentSchema],
+    user: UserSchema,
+    created: {
```
Comment (ryan-hamblin) [inline api/pullrequests/pr.model.js]: I'm wondering if we want to create our own dates, or use the timestamp that Github Gives us? Trying to figure out which would be better.

File: api/utils/utilities.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+module.exports = () => {
```
Comment (ryan-hamblin) [inline api/utils/utilities.js:1]: Any type of handler should probably be saved here in the utils file. good stuff!

**Review by ryan-hamblin — COMMENTED** [review]
Great job here. I would like to clean a few things up. Indentation/file nameing and maybe in the future put all those helpers int he utils file.

File: api/pullrequests/pr.controller.js (hunk 5 above)
Comment (swoleengineer) [reply api/pullrequests/pr.controller.js]: Yes sir, I figured they had to have a way to help us decipher what is going on.lol

File: api/home/home.controller.js (hunk 1 above)
Comment (swoleengineer) [reply api/home/home.controller.js]: Definitely will do, I never usually pay attention to my indentation/spacing.

File: api/pullrequests/pr.model.js (hunk 6 above)
Comment (swoleengineer) [reply api/pullrequests/pr.model.js]: It would almost be the same though right?

**Pull request merged**